Python/pystate.c

```diff
@@ -3002,20 +3002,20 @@ _PyThreadState_CheckConsistency(PyThreadState *tstate)
 //
 // tstate can be a dangling pointer (point to freed memory): only tstate value
 // is used, the pointer is not deferenced.
-//
-// tstate must be non-NULL.
 int
 _PyThreadState_MustExit(PyThreadState *tstate)
 {
+    // tstate must be non-NULL.
+    if (tstate == NULL) {
+        return 1;
+    }
```

Comment:
Let's just make this an assertion.
```suggestion
    assert(tstate != NULL);
```